.github/workflows/backport.yml

```diff
@@ -70,8 +75,8 @@ jobs:
               ]
             }
         env:
-          SLACK_WEBHOOK_URL: ${{ secrets.BACKPORT_ASSISTANT_FAILURE_SLACK }}
+          SLACK_WEBHOOK_URL: ${{ env.SLACK_FEED_NOMAD || secrets.BACKPORT_ASSISTANT_FAILURE_SLACK }}
           SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK
 permissions:
   contents: read
-
+  id-token: write
```

Comment:
What's this change?

Reply:
high-level explanation is that the [`vault-auth` helper script](https://github.com/hashicorp/nomad/pull/17841/files#diff-2ccd1b5860617862a60112949549f7739cf06ecc92cc2511d60dcbdd5c96e4d6R13) on our self-hosted runners needs it to provide a Vault token based on the repo of the workflow run.

longer explanation of GHA's OIDC setup is hereabouts: https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect#updating-your-actions-for-oidc
> The `id-token: write` setting allows the JWT to be requested from GitHub's OIDC provider [...]

I could not find a way to make this conditional for ent-only -- it must be a static string.